keps/0007-kic-performance-test.md

```diff
@@ -0,0 +1,557 @@
+---
+title: KIC Performance Test Solution
+status: implementable
+---
+
+# Notes
+For references see the dependencies related to this proprosal to check the progress of related efforts:
+- solution 
+- deployment and configuration
+- code/repo
+- configuration and execution
+- profiling (code profiler)
+- histogram execution time and calculates percentiles
+
+
+## Summary
+  We want a performance test solution that can collect controller metrics per release, which also provides profiling
+and histogram execution time. The solution is able to scale according to controller scalability. We mainly use kind as test cluster, but also support varied cloud provider cluster such EKS, AKS, GKE etc.
```

Comment:
I have a slightly different idea of what our summary should be based on some desire to use language that makes things more generic (and less specific to our use cases):

```suggestion

Testing libraries and tools for reporting performance metrics of the Kong Kubernetes Ingress Controller (KIC) in any Kubernetes cluster environment.
```

Let me know your thoughts, and whether this wording resonates with you or not?

Reply:
A more generic description is : we need testing tools/framekwork such as jenkins, KIC, or curl finish testing, and test report could be publicly accessible downloadable.

Reply:
Personally I think pointing out specific tools in this doc (especially this iteration) makes us _less_ generic. I would like to see a truly high level summary that states the basic thing we want to accomplish with as few details as necessary.

Reply:
updated accordingly.

Reply:
I've read the new one, this reads well for me so let's go with this for now :+1: